tests/llm/test_model_config.py

```
@@ -0,0 +1,351 @@
+        EMBEDDING=EmbeddingSettings(VECTOR_DIMENSIONS=2048),
+        VECTOR_STORE=VectorStoreSettings(TYPE="lancedb", MIGRATED=True),
+    )
+
+    assert settings.EMBEDDING.VECTOR_DIMENSIONS == 2048
+    assert settings.VECTOR_STORE.DIMENSIONS == 2048
+
+
+def test_app_settings_require_matching_embedding_and_vector_store_dimensions() -> None:
+    with pytest.raises(
+        ValueError,
+        match="VECTOR_STORE.DIMENSIONS must match EMBEDDING.VECTOR_DIMENSIONS",
```

Ruff (0.15.9)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)